# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
# Commit: 606555f7b50c
# Full Hash: 606555f7b50cae920f46328d31eb07b50557d381
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2023-04-18 09:19:34
# Description:
#   Bug 1825514 - In VideoCaptureAvFoundation, lock the monitor before notifying it. r=webrtc-reviewers,jesup,dbaker
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D175654
# ==============================================================================

diff -r 7d0ed4ed1f56 -r 606555f7b50c dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm
--- a/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Mon Apr 17 14:46:29 2023 +0000
+++ b/dom/media/systemservices/objc_video_capture/video_capture_avfoundation.mm	Mon Apr 17 14:51:01 2023 +0000
@@ -158,6 +158,7 @@
                                format:format
                                   fps:aCapability.maxFPS
                     completionHandler:^(NSError* error) {
+                      MonitorAutoLock lock2(*copyableMonitor);
                       MOZ_RELEASE_ASSERT(!rv);
                       rv = Some(error ? -1 : 0);
                       copyableMonitor->Notify();
@@ -224,6 +225,7 @@
   __block bool done = false;
 
   [mCapturer stopCaptureWithCompletionHandler:^(void) {
+    MonitorAutoLock lock2(*copyableMonitor);
     MOZ_RELEASE_ASSERT(!done);
     done = true;
     copyableMonitor->Notify();
